Firestore/Source/Local/FSTLocalDocumentsView.mm

```diff
@@ -62,21 +62,32 @@ - (instancetype)initWithRemoteDocumentCache:(id<FSTRemoteDocumentCache>)remoteDo
 }
 
 - (nullable FSTMaybeDocument *)documentForKey:(const DocumentKey &)key {
+  NSArray<FSTMutationBatch *> *affectingBatches =
+      [self.mutationQueue allMutationBatchesAffectingDocumentKey:key];
+  return [self documentForKey:key withAffectingBatches:affectingBatches];
+}
+
+// Internal version of documentForKey: which allows reusing `affectingBatches`.
+- (nullable FSTMaybeDocument *)documentForKey:(const DocumentKey &)key
+                         withAffectingBatches:(NSArray<FSTMutationBatch *> *)affectingBatches {
```

Comment:
In general avoid prepositions in secondary arguments unless their meaning is required to understand the behavior. A good usage would be `copyFromFile:toFile:`. This doesn't need "with".

Also, "affecting" really is part of the function that looks up the mutations in the mutation queue, but after that point they're just batches. Taken together I'd suggest renaming this to `documentForKey:inBatches:`.

Similarly I'd suggest renaming locals of the form `affectingBatches` to just `batches` or `matchingBatches`.